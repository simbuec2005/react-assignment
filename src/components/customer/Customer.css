.customerRow {
    display: flex;
    border-bottom: 1px solid #ccc;
    border-right: 1px solid #ccc;
    border-left: 1px solid #ccc;
    cursor: pointer;
}

.customerRow:hover {
    background-color: #d6d6d6;
}

.customerRow div {
    padding: 10px;
    flex: 1;
}