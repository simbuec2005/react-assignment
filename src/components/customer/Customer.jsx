import React from 'react';
import './Customer.css';
class Customer extends React.Component {

    render() {

        const { detail, handleClick } = this.props;
        return (
            <div className="customerRow" onClick={() => {
                handleClick(detail)
            }}>
                <div>{detail.id}</div>
                <div>{detail.name}</div>
                <div>{detail.age}</div>
            </div>
        )
    }

}

export default Customer;