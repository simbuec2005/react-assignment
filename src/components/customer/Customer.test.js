import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import Customer from './Customer';

const store = createStore(() => { return {}; });

describe('<Customer />', () => {
  test('Should match snapshot', () => {
    const props = {
      detail: {
        "id": "123",
        "age": "25",
        "name": "Neil Amstrong"
      },
      handleClick: ()=> {}
    }
    const component = create(
      <Customer {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
})
