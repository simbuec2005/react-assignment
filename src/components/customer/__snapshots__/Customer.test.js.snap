// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Customer /> Should match snapshot 1`] = `
<div
  className="customerRow"
  onClick={[Function]}
>
  <div>
    123
  </div>
  <div>
    Neil Amstrong
  </div>
  <div>
    25
  </div>
</div>
`;
